source/Halibut/Transport/Protocol/MessageSerializer.cs

```diff
@@ -75,8 +75,9 @@ public async Task WriteMessageAsync<T>(Stream stream, T message, CancellationTok
         {
             using var compressedByteCountingStream = new ByteCountingStream(stream, OnDispose.LeaveInputStreamOpen);
             using var compressedInMemoryBuffer = new WriteIntoMemoryBufferStream(compressedByteCountingStream, writeIntoMemoryLimitBytes, OnDispose.LeaveInputStreamOpen);
+            using var apmToTapStream = new ApmToTapStream(compressedInMemoryBuffer);
```

Comment:
This stream will have dispose call dispose on the inner stream, which is not inline with what is happening on the other streams but will be fine since the other streams I think will prevent that dispose call from making it all the way down.